Src/Fare/State.cs

```diff
@@ -157,6 +157,33 @@ public override int GetHashCode()
         }
 
         /// <summary>
+        /// Compares the current instance with another object of the same type
+        /// and returns an integer that indicates whether the current instance precedes,
+        /// follows, or occurs in the same position in the sort order as the other object.
+        /// </summary>
+        /// <param name="other">An object to compare with this instance.</param>
+        /// <returns>
+        /// A value that indicates the relative order of the objects being compared.
+        /// The return value has these meanings: Value Meaning Less than zero This instance precedes <paramref name="other" /> in the sort order.
+        /// Zero This instance occurs in the same position in the sort order as <paramref name="other" />.
+        /// Greater than zero This instance follows <paramref name="other" /> in the sort order.
+        /// </returns>
+        public int CompareTo(object other)
+        {
+            if (other == null)
+            {
+                return 0;
```

Comment:
To be honest, I find `IComperable` to be just a bad design, because returning some `int` just doesn't make sense, unless you've read how the protocol works... :confused: